src/resources/app/pages/course/Announcement.vue

```diff
@@ -10,6 +10,11 @@ import AppLayout from "../../components/Layout/AppLayout";
 
 export default {
   layout: [AppLayout, CourseStream],
-  components: {AnnouncementContainer}
+  components: {AnnouncementContainer},
+  beforeCreate() {
+    if (!this.$store.state.isAuthenticated) {
+      window.location.assign("/login");
+    }
+  }
```

Comment:
I think you should add some props in this page, so you could fetch param 'id' from URL